research/management/commands/orcid-researcher-import.py

```diff
@@ -0,0 +1,231 @@
+
+        self.get_orcid_ids()
+        self.match_teledata_records()
+        self.print_stats()
+
+    def get_orcid_ids(self):
+        """
+        Gather all the ORCID IDs for the provided
+        institution GRID ID or name.
+        """
+        request_url = '{0}{1}'.format(
+            settings.ORCID_BASE_API_URL,
+            'search/'
+        )
+
+        params = {}
+        headers = {
+            'Accept': 'application/json'
+        }
+
+        if self.use_grid:
+            params['q'] = 'grid-org-id:{0}'.format(self.grid_id)
+        else:
+            params['q'] = 'affiliation-org-name:("{0}")'.format(self.institution_name)
+
+        data = self.__request_records(0, request_url, params, headers)
+
+        for row in data['result']:
+            self.orcid_ids_to_import.append(row['orcid-identifier']['path'])
+            self.processed += 1
+
+        self.total_records = data['num-found']
+        total_pages = math.ceil(self.total_records / 100)
+        current_page = 1
+
+        while (current_page <= total_pages):
+            data = self.__request_records(current_page, request_url, params, headers)
+
+            for row in data['result']:
+                self.orcid_ids_to_import.append(row['orcid-identifier']['path'])
+                self.processed += 1
+
+            current_page += 1
+
+        self.stdout.write('Collected {0} of {1} ORCID records found...\n'.format(self.processed, self.total_records))
+
+
+    def match_teledata_records(self):
+        """
+        Loop through the found ORCID IDs,
+        gather more information and attempt
+        to match them up with our teledata
+        records.
+        """
+        headers = {
+            'Accept': 'application/json'
+        }
+
+        self.progress_bar = ChargingBar(
+            'Processing',
+            max=len(self.orcid_ids_to_import)
+        )
+
+        for idx, orcid_id in enumerate(self.orcid_ids_to_import):
+            self.progress_bar.next()
+            try:
+                Researcher.objects.get(
+                    orcid_id=orcid_id
+                )
+
+                # Not really, but it makes for better
+                # at-a-glance understanding
+                self.updated += 1
+                # Nothing to do, continue
+                continue
+            except Researcher.DoesNotExist:
+                existing = None
+
+            request_url = "{0}{1}".format(
+                settings.ORCID_BASE_API_URL,
+                orcid_id
+            )
+
+            data = self.__request_records(0, request_url, {}, headers)
+
+            try:
+                first_name = data['person']['name']['given-names']['value']
+                last_name = data['person']['name']['family-name']['value']
+            except KeyError:
+                self.stdout.write("Unable to get first or last name from record.\n")
+                self.error += 1
+                continue
+            except TypeError:
```

Comment:
Unique error messages for these particular types of errors could be helpful--alternatively, we could combine the two with something like `except (KeyError, TypeError):`

Reply:
Combined the two exceptions. I don't think we really care which one is thrown. In either case, there isn't a first or last name for us to process.